trinity/sync/full/state.py

```diff
@@ -245,11 +221,42 @@ def deregister_peer(self, peer: BasePeer) -> None:
                 return
 
             candidates = list(not_yet_requested.difference(self._peer_missing_nodes[peer]))
-            batch = candidates[:eth_constants.MAX_STATE_FETCH]
+            batch = tuple(candidates[:eth_constants.MAX_STATE_FETCH])
             not_yet_requested = not_yet_requested.difference(batch)
             self.request_tracker.active_requests[peer] = (time.time(), batch)
-            self.logger.debug("Requesting %d trie nodes to %s", len(batch), peer)
-            peer.sub_proto.send_get_node_data(batch)
+            asyncio.ensure_future(self._request_and_process_nodes(peer, batch))
+
+    async def _request_and_process_nodes(self, peer: ETHPeer, batch: Tuple[Hash32, ...]) -> None:
+        self.logger.debug("Requesting %d trie nodes from %s", len(batch), peer)
+        node_data = await peer.requests.get_node_data(batch)
+        self.request_tracker.active_requests.pop(peer)
```

Comment:
How do we know there's no `KeyError` here?
Why store the `time.time()` just to drop it?

Reply:
> How do we know there's no KeyError here?

We don't.  I'll add some exception handling here for the interim until this API gets removed entirely.  My plan is to do another round of refactoring on the `StateDownloader` to remove this `request_tracker` API because I don't believe it is needed anymore.